pkg/config/sysprep_test.go

```diff
@@ -83,35 +84,34 @@ var _ = Describe("SysprepConfigMap", func() {
 		},
 	})
 
-	Describe("With invalid file name", func() {
-		BeforeEach(func() {
-			createFile(filepath.Join(SysprepSourceDir, "sysprep-volume", "wrongname.xml"))
-		})
+	DescribeTable("Assert CreateSysprepDisks behaviour",
+		func(vmi *v1.VirtualMachineInstance, filenames []string, shouldSucceed bool) {
 
-		It("Should fail on creating config map iso disk", func() {
-			err := CreateSysprepDisks(vmiConfigMap, false)
-			Expect(err).To(HaveOccurred())
-		})
-	})
+			for _, f := range filenames {
+				createFile(filepath.Join(SysprepSourceDir, "sysprep-volume", f))
+			}
 
-	Describe("With valid file name (autounattend.xml)", func() {
-		BeforeEach(func() {
-			// Check case-insensitivity (should accept anything Windows accepts).
-			createFile(filepath.Join(SysprepSourceDir, "sysprep-volume", "AutounattenD.xml"))
-		})
-
-		It("Should create a new config map iso disk", func() {
-			err := CreateSysprepDisks(vmiConfigMap, false)
-			Expect(err).NotTo(HaveOccurred())
-			_, err = os.Stat(filepath.Join(SysprepDisksDir, "sysprep-volume.iso"))
-			Expect(err).NotTo(HaveOccurred())
-		})
-
-		It("Should create a new secret iso disk", func() {
-			err := CreateSysprepDisks(vmiSecret, false)
-			Expect(err).NotTo(HaveOccurred())
-			_, err = os.Stat(filepath.Join(SysprepDisksDir, "sysprep-volume.iso"))
-			Expect(err).NotTo(HaveOccurred())
-		})
-	})
+			if shouldSucceed {
```

Comment:
The usage of a boolean as an argument usually hints of a smell.
I think the inner condition here shows the problem: The test body is actually split by the boolean, and the repetition it is suppose to avoid is minor.

The original fixture in which the files got created before the test, seem more appropriate to me.

You could have two tables here, one for success and one for failure, I think that will solve this nicely.

Reply:
I'd agree about the code smell outside of unit tests...

The repetition I'm trying to avoid can be seen in a previous revision of this commit:

https://github.com/kubevirt/kubevirt/commit/50026e6556d51622a023c3bfe134a2161f244c97

I can add another DescribeTable for the negative tests if really required but that seems pointless tbh.

I'm also not sure what your point is regarding file creation, we can't dynamically create these in BeforeEach when using DescribeTable AFAICT but I'm likely missing something.

Reply:
> I can add another DescribeTable for the negative tests if really required but that seems pointless tbh.

It will drop the need for the boolean and will structure the tests nicely. I'm not sure if I miss something else that may make this less attractive.

> I'm also not sure what your point is regarding file creation, we can't dynamically create these in BeforeEach when using DescribeTable AFAICT but I'm likely missing something.

You are right, sorry. It is based on the input.
There are two options:
- Leave it as is and possibly pass it to ta helper function (e.g. `createFiles(...)`).
- Pass in a function that does this. The function is constructed in the entry.
  e.g. `Entry(..., func(){createFiles([]string{"my.xml"})}),`
  
IMO the 1st simple option is good enough.

Reply:
Okay I've added the second table and extended `createFile` as suggested.